project/Build.scala

```diff
@@ -322,13 +332,26 @@ object Build extends sbt.Build {
       }
   )
 
-  /** Depend library as if (exportJars in library) was set to false */
-  val compileWithLibrarySetting = {
-    internalDependencyClasspath in Compile ++= {
-      val prods = (products in (library, Compile)).value
-      val analysis = (compile in (library, Compile)).value
+  implicit class ProjectOps(val project: Project) extends AnyVal {
+    /** Depends on the compiler */
+    def dependsOnCompiler: Project =
```

Comment:
Consider calling this `withScalaJSCompiler` or something. If we already have a dedicated method, it seems to me that "depending" is not the best term.